client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/RemoveIndexLifecyclePolicyRequest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.indexlifecycle;
+
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.client.TimedRequest;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class RemoveIndexLifecyclePolicyRequest extends TimedRequest implements IndicesRequest.Replaceable {
+
+    private String[] indices;
+    private IndicesOptions indicesOptions = IndicesOptions.strictExpandOpen();
+
+    public RemoveIndexLifecyclePolicyRequest() {
+    }
+
+    public RemoveIndexLifecyclePolicyRequest(String... indices) {
+        if (indices == null) {
+            throw new IllegalArgumentException("indices cannot be null");
+        }
+        this.indices = indices;
+    }
+
+    @Override
+    public RemoveIndexLifecyclePolicyRequest indices(String... indices) {
```

Comment:
This is a question, not a change request: What is our philosophy regarding having setters vs. immutable request objects going forward for the HLRC? I've been under the impression we preferred immutable objects, but it doesn't seem to be consistent.

Reply:
@gwbrown Good question that I would like to know the answer to myself :).  I was mostly going off what I saw within the set index lifecycle policy classes since that's the analog to this request.

Reply:
I think the most important thing is for the API to be built in a way where you cannot create invalid requests. It would be good if any mandatory fields in the request were passed into the constructor and don't have setters. The reason this was done for the server side was because the request extends `IndicesRequest.Replaceable` but I think in order to achieve the client separation we want we actually can't extend that here?

Reply:
yea Im all for not exetnding that class. And Im also all for putting things that are primitive and easily validatable into the constructors. Optionals, i think im ok with setters but i think this also deserves a wider audience to discuss.

Reply:
Removed the extension.  However, IndicesOptions as a member variable still seems necessary.  There is no way to specify how to handle wild cards appropriately without it, and the SetIndexLifecyclePolicyRequest also uses them.